com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -366,62 +383,107 @@ internal Vector3 Scale
         /// </summary>
         protected NetworkManager m_CachedNetworkManager;
 
-        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkState = new NetworkVariable<NetworkTransformState>(new NetworkTransformState());
+        /// <summary>
+        /// We have two internal NetworkVariables.
+        /// One for server authoritative and one for "client/owner" authoritative.
+        /// </summary>
+        private readonly NetworkVariable<NetworkTransformState> m_ReplicatedNetworkStateServer = new NetworkVariable<NetworkTransformState>(new NetworkTransformState(), NetworkVariableReadPermission.Everyone, NetworkVariableWritePermission.Server);
```

Comment:
Since the difference between these two is only a matter of what parameters are passed to the constructor, and from what I can tell only one of them is ever used, couldn't these just be collapsed to just one that's initialized in OnNetworkSpawn? Granted as it's currently implemented we have no way to react to a change from server- to owner-authoritative... but that seems to be a problem in the current implementation, too, since changing that would suddenly change it to using the other one that's probably gonna be all 0s?

Reply:
You can't initialize NetworkVariables in OnNetworkSpawn.  By then it is too late... they have to be initialized either when they are declared or in Awake.  Since user code could declare an Awake within a child derived class, this would break the initialization since the NetworkTransform.Awake would not be invoked.  They could also change a property value returned in the IsServerAuthoritative in their Start method.

It is the safest way to do this for the patch.